lib/matplotlib/ticker.py

```diff
@@ -207,534 +212,399 @@ def get_tick_space(self):
+    """Use a new-style format string (as used by `str.format()`) to
+    format the tick.  The field formatting must be labeled `x` and/or
+    `pos`.
     """
 
-    def __call__(self, x, pos=None):
-        'Return the format for tick val *x* at position *pos*'
-        xmin, xmax = self.axis.get_view_interval()
-        d = abs(xmax - xmin)
-
-        return self.pprint_val(x, d)
-
-    def pprint_val(self, x, d):
-        #if the number is not too big and it's an int, format it as an
-        #int
-        if abs(x) < 1e4 and x == int(x):
-            return '%d' % x
+    def __init__(self, fmt):
+        super(StrMethodFormatter, self).__init__()
+        self._fmt = fmt
 
-        if d < 1e-2:
-            fmt = '%1.3e'
-        elif d < 1e-1:
-            fmt = '%1.3f'
-        elif d > 1e5:
-            fmt = '%1.1e'
-        elif d > 10:
-            fmt = '%1.1f'
-        elif d > 1:
-            fmt = '%1.2f'
-        else:
-            fmt = '%1.3f'
-        s = fmt % x
-        #print d, x, fmt, s
-        tup = s.split('e')
-        if len(tup) == 2:
-            mantissa = tup[0].rstrip('0').rstrip('.')
-            sign = tup[1][0].replace('+', '')
-            exponent = tup[1][1:].lstrip('0')
-            s = '%se%s%s' % (mantissa, sign, exponent)
-        else:
-            s = s.rstrip('0').rstrip('.')
-        return s
+    def format_for_tick(self, x, pos=None):
+        return self._fmt.format(x=x, pos=pos)
 
 
 class ScalarFormatter(Formatter):
-    """
-    Tick location is a plain old number.  If useOffset==True and the data range
-    is much smaller than the data average, then an offset will be determined
-    such that the tick labels are meaningful. Scientific notation is used for
-    data < 10^-n or data >= 10^m, where n and m are the power limits set using
+    """Tick location is a plain old number.  If use_offset==True
+    and the data range is much smaller than the data average, then
+    an offset will be determined such that the tick labels are
+    meaningful. Scientific notation is used for data < 10^-n or
+    data >= 10^m, where n and m are the power limits set using
     set_powerlimits((n,m)). The defaults for these are controlled by the
     axes.formatter.limits rc parameter.
-
     """
 
-    def __init__(self, useOffset=None, useMathText=None, useLocale=None):
-        # useOffset allows plotting small data ranges with large offsets: for
-        # example: [1+1e-9,1+2e-9,1+3e-9] useMathText will render the offset
-        # and scientific notation in mathtext
-
-        if useOffset is None:
-            useOffset = rcParams['axes.formatter.useoffset']
-        self.set_useOffset(useOffset)
-        self._usetex = rcParams['text.usetex']
-        if useMathText is None:
-            useMathText = rcParams['axes.formatter.use_mathtext']
-        self._useMathText = useMathText
-        self.orderOfMagnitude = 0
-        self.format = ''
+    def __init__(self, *args, **kwargs):
```

Comment:
Please do not use kwarg popping and please preserve the current API (`useMathText`).